src/lib.rs

```diff
@@ -108,8 +112,11 @@ fn _internal(py: Python, m: &PyModule) -> PyResult<()> {
     m.add_submodule(store)?;
 
     // Register substrait as a submodule
+    #[cfg(feature = "substrait")]
     let substrait = PyModule::new(py, "substrait")?;
+    #[cfg(feature = "substrait")]
     substrait::init_module(substrait)?;
+    #[cfg(feature = "substrait")]
     m.add_submodule(substrait)?;
```

Comment:
nit: we could move this code into a function and then have just one `#[cfg]` directive when calling that function